### DISCUSSION TITLE
Nuances on build strategies, clarification needed.

### DISCUSSION THREAD

**Discussion by afidegnum** [framesurge/perseus#240] [General]
hi, @arctic-hen7, 

With the help of sycamore/perseus, I'm able to have a full working prototype and believe me, these framework will go far in getting a massive adoption. 

I'm able to have this [Prototype working](https://linksapp.flashtechhub.com/) ready.

I encountered few situations I need your input. 

I needed to visit a path with custom dynamic parameter (`path/<param>`), in my case, I needed to query a link from the remote api, in order to fetch it's detailed result, 

What I did, still not sure if it's the correct one, was to load the list of path into the `build_path` as you can see at [this repo](https://github.com/commune-org/linksapp-proto/blob/main/front/src/templates/linkpage.rs#L139)

however on `incremental_generation()` I had a 502 error which I had to disable and further investigate 

I also would like to know when to use `request_state_fn` I thought it would serve as a replacement of i.e `reqwest` or any other http client. can you please clarify on that aspect?

**Comment by arctic-hen7** [comment]
Seems like you're using build paths correctly, but I'm not sure why incremental would fail with a 502. What is the exact error message you're getting?

Request state isn't an HTTP client, it's just like build state, except it runs at the time a user makes a request. That means you can generate state that's specific to details about their request. Generally, build state and incremental generation should be enough, but if you're handling authentication, you might need request state. Also, state amalgamation let's you combine request state with build state.

**Reply by afidegnum** [reply]
Thanks a lot, for the insight,
